1011/en.md

```diff
@@ -0,0 +1,85 @@
+Here 1,2,4 denotes first,second,third woman respectively.Now if 1st man marries the 1st woman we will make the mask `(mask|(1<<0))==1)`.It  means the woman number 1 is married with 1st man but if we made `(mask|(1<<1)==2)` it means the first man married the 2nd woman.This way we will have to consider all the jth woman to find the greatest maximum sum of pririties for the `ith` man
+Now suppose the 2nd man marries 2nd woman then we make the `mask|(1<<1)==1|2==3`.Now if you know the "Bit Manipulation Technique" you will know `(1<<1)==2^1=2` and you know 2 means taking 2nd woman into consideration and masking 2 with the previous 1 makes it 3 means both 1st and 2nd woman are married and they can't marry any other man neither any man can consider marrying them either
+Now since 1st woman and 2nd woman are taken the mask is 3==011 . Now for 3rd woman we can't take the either "1s" of the binary number.So the only option left for 3rd man is the 3rd woman which is the 3rd bit from the right which is still zero
+Now choosing the 3rd bit that is the 3rd woman we make `mask|(1<<2)==3|4==7` that is, we considered all the man and woman without any overlapping and so we completed a combination and so we return 0.And how we ignored the overlapping?For the ith man we simply avoided woman or bit which is set as `1` and we condered the bits which are `0` and considered all combinations we can take and stored the maximum inside a variable
+That maximum number which is a combination is the answer.The number of total combinations is `2^N-1` and we will have to compute all of them and return the maximum one out there which will denote the maximum sum of priority.
+
+The recursive formula of this problem is:
+`
+ bitmask(i,(1<<n)-1)=0 
+ bitmask(i,mask)=max(a[i][j]+bitmask(i+1,mask|(1<<j)) where i<=n,j<=n and a[i][j] consists of priority of ith man for jth woman
+`
+
+### Resources:
+Here are some resources of  **Bit Manipulation,Basic Dp and BitMask Dp** so that you can have a deeper understanding and learn to solve problems using these resources:-
+
+- [Shafaetsplanet](https://www.shafaetsplanet.com/?p=1357) **(BitMask Dp)**
+- [HackerEarth](https://www.hackerearth.com/practice/basic-programming/bit-manipulation/basics-of-bit-manipulation/tutorial/) **(Basics of Bit Manipulation)**
+- [Bangladesh Advanced Computing Society - BACS](https://www.youtube.com/watch?v=cbgdSX2pXcQ)   **(BitMask Dp)**
+- [GeeksforGeeks](https://www.geeksforgeeks.org/dynamic-programming/)   **(Basic and Advanced Types of DP)**
+
+
+### C++ Implementation:
+```
+#include<bits/stdc++.h>
+using namespace std;
+int a[16][16];
+int dp[1<<16];//1<<16==2^16 since in worst case there can be 16 marriage ceremonies
+int n;
+int bitmask(int i,int mask)
+{
+    if(mask==(1<<n)-1) //base case
+        return 0;
+    int &ans=dp[mask];//& is a reference operator and any change brought to a reference variable will be set to the dp array at mask index because of this operator
```

Comment:
The explanation regarding the reference operator is unnecessary here. So, please remove it.